src/Pos/Lrc/Worker.hs

```diff
@@ -142,23 +145,23 @@ lrcDo getPeers epoch consumers tip = tip <$ do
         Nothing -> throwM UnknownBlocksForLrc
         Just (NewestFirst -> blunds) ->
             withBlocksRolledBack blunds $ do
-                -- We have to compute richmen before calculating the seed because
-                -- seed calculation relies on knowing richmen
                 issuersComputationDo epoch
-                richmenComputationDo epoch consumers
+                -- We have to compute richmen before calculating the seed
+                -- because seed calculation relies on knowing richmen. We
+                -- give 'calcAndStoreSeed' as a callback to
+                -- 'richmenComputationDo' because it's easier than reading
+                -- richmen from the database (I couldn't find a function that
+                -- does it on a locked LRC database)
+                let simpleConsumer = lrcConsumerFromComponentSimple @RCSsc bvdMpcThd
+                let calcSeedConsumer = simpleConsumer {
+                        lcComputedCallback = calcAndStoreSeed }
+                richmenComputationDo epoch (calcSeedConsumer:consumers)
```

Comment:
I don't understand this part. When we calculate seed for the 10-th epoch we use data from the 9-th epoch. So we should use richmen from the 9-th epoch, don't we? In this case all this stuff can be somewhat simplified I suppose.